modern/mdlogger/datadog.go

```diff
@@ -0,0 +1,105 @@
+package mdlogger
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/short-d/app/fw"
+)
+
+// DataDog logging API =>
+//   https://docs.datadoghq.com/api/?lang=bash#logs
+const dataDogLoggingApi = "https://http-intake.logs.datadoghq.com/v1/input"
+
+var _ EntryRepository = (*DataDogEntryRepo)(nil)
+
+type DataDogEntryRepo struct {
+	apiKey      string
+	httpRequest fw.HTTPRequest
+	env         fw.ServerEnv
+}
+
+func (d DataDogEntryRepo) createLogEntry(
+	level fw.LogLevel,
+	prefix string,
+	line int,
+	filename string,
+	message string,
+	date time.Time) {
+	headers := d.authHeaders()
+
+	body := d.requestBody(level, prefix, line, filename, message)
+	jsonBody, err := json.Marshal(body)
+	if err != nil {
+		return
+	}
+	var res = make(map[string]interface{})
+	_ = d.httpRequest.JSON(http.MethodPost, dataDogLoggingApi, headers, string(jsonBody), &res)
+}
+
+func getSeverity(level fw.LogLevel) string {
+	switch level {
+	case fw.LogFatal:
+		return "critical"
+	case fw.LogError:
+		return "error"
+	case fw.LogWarn:
+		return "warning"
+	case fw.LogInfo:
+		return "info"
+	case fw.LogDebug:
+		return "debug"
+	case fw.LogTrace:
+		return "debug"
```

Comment:
trace?

Reply:
@arberiii We are mapping to DataDog severities: https://docs.datadoghq.com/logs/processing/processors/?tab=ui#log-status-remapper